services/QuillLMS/client/app/bundles/Staff/components/evidence/configureRules/rulePlagiarismAttributes.tsx

```diff
@@ -11,23 +12,60 @@ import { ClickEvent } from '../../../interfaces/evidenceInterfaces';
 
 // TODO: add props interface
 
-export const PlagiarismTextEditor = ({ text, index, setPlagiarismText, }) => {
+export const PlagiarismTextEditor = ({ text, index, setPlagiarismText, id, indexToShow, }) => {
   function onHandleSetPlagiarismText(text) {
     setPlagiarismText(text, index)
   }
 
   return (
     <React.Fragment key={index}>
-      <p className="form-subsection-label">Plagiarism Text - Text String {index + 1}</p>
+      <p className="form-subsection-label">Plagiarism Text - Text String {indexToShow}</p>
       <TextEditor
         handleTextChange={onHandleSetPlagiarismText}
-        key={`plagiarism-text-${index}`}
+        key={`plagiarism-text-${id}`}
         text={text}
       />
     </React.Fragment>
   )
 }
 
+export const PlagiarismText = ({ onHandleSetPlagiarismText, plagiarismText, plagiarismTextIndex, setPlagiarismTexts, addPlagiarismTextButton, plagiarismTexts }) => {
+  function handleRemovePlagiarismText() {
+    if (window.confirm('Are you sure you want to remove this highlight?')) {
+      if (plagiarismText.id) {
+        const newPlagiarismTexts = [...plagiarismTexts]
+        newPlagiarismTexts[plagiarismTextIndex] = { ...plagiarismText, _destroy: true }
+        setPlagiarismTexts(newPlagiarismTexts)
+      } else {
+        const newPlagiarismTexts = plagiarismTexts.filter((_, index) => index !== plagiarismTextIndex);
+        setPlagiarismTexts(newPlagiarismTexts);
+      }
+    }
+  }
+
+  const visiblePlagiarismTexts = plagiarismTexts.filter(pt => !pt._destroy)
+  const indexToShow = visiblePlagiarismTexts.findIndex(pt => plagiarismText.id ? pt.id === plagiarismText.id : pt.tempId === plagiarismText.tempId) + 1
```

Comment:
This `tempId` approach feels a little bit awkward since you have to special case everywhere we access `.id`, but I can't actually think of a cleaner way to do this off the top of my head, so it seems like a reasonable solution.